File: dotnet/WebService/src/WebApi/Dtos/Base64Dto.cs
using WebApi.Utils;

namespace WebApi.Dtos
{
    public class Base64Dto
    {
        public string Message { get; set; }

        public string EncodedMessage { get; set; }

        public string DecodedMessage { get; set; }

        public Base64Dto()
        {
        }

        public Base64Dto(string message)
        {
            Message = message;
            EncodedMessage = Base64.Encode(Message);
            DecodedMessage = Base64.Decode(EncodedMessage);
        }
    }
}

File: laravel/webservice/app/Dtos/Base64Dto.php
<?php

namespace App\Dtos;

use App\Utils\Base64;

class Base64Dto
{
    public string $message;
    public string $encodedMessage;
    public string $decodedMessage;

    public function __construct($message)
    {
        $this->message = $message;
        $this->encodedMessage = Base64::encode($this->message);
        $this->decodedMessage = Base64::decode($this->encodedMessage);
    }
}
